src/app/shared/services/cart.service.ts

```diff
@@ -67,72 +67,45 @@ export class CartService {
-        }).subscribe(
-          (data) => {
-            item.quantity = value.number || 1;
-            this.cartItems.next(this.cartItems.getValue());
-          },
-          (err) => {
-            console.error('Cannot update (reduce quantity) in server: ', err);
-          }
-        );
-      } else {
-        // Modify in localStorage
-        item.quantity = value.number || 1;
-        localStorage.setItem(this.localStorageKey, JSON.stringify(this.cartItems.getValue()));
-        this.cartItems.next(this.cartItems.getValue());
-      }
-    } else {
-      // Should delete and add new product's instance
-      if (this.authService.isLoggedIn.getValue()) {
-        // Change in server
-        this.httpService.post('order/delete', {
-          product_instance_id: value.pre_instance_id,
-        }).subscribe(
-          (data) => {
-            this.httpService.post('order', {
-              product_id: value.product_id,
-              product_instance_id: value.instance_id,
-              number: value.number
-            }).subscribe(
-              (dt) => {
-                this.getCartItems();
-              },
-              (err) => {
-                console.error('Cannot add new order-line to order in server: ', err);
-              }
-            );
-          },
-          (err) => {
-            console.error('Cannot delete the specific product instance in server: ', err);
-          }
-        );
-      } else {
-        // Change in localStorage
-        let ls_items = this.getItemsFromStorage();
-        ls_items = ls_items.filter(el =>
-          el.product_id !== value.product_id ||
-          el.instance_id !== value.pre_instance_id);
-
-        for (let counter = 0; counter < value.number; counter++)
-          ls_items.push({
+    // Should delete and add new product's instance
+    if (this.authService.isLoggedIn.getValue()) {
+      // Change in server
+      this.httpService.post('order/delete', {
+        product_instance_id: value.pre_instance_id,
+      }).subscribe(
+        (data) => {
+          this.httpService.post('order', {
             product_id: value.product_id,
-            instance_id: value.instance_id,
-          });
+            product_instance_id: value.instance_id,
+            number: value.number
+          }).subscribe(
+            (dt) => {
+              this.getCartItems();
+            },
+            (err) => {
+              console.error('Cannot add new order-line to order in server: ', err);
+            }
+          );
+        },
+        (err) => {
+          console.error('Cannot delete the specific product instance in server: ', err);
+        }
+      );
+    } else {
+      // Change in localStorage
+      let ls_items = this.getItemsFromStorage();
+      ls_items = ls_items.filter(el =>
+        el.product_id !== value.product_id ||
+        el.instance_id !== value.pre_instance_id);
+
+      for (let counter = 0; counter < value.number; counter++)
```

Comment:
In some part the number of order lines is stored in localStorage but here you save order lines individualy